services/queue/src/api.js

```diff
@@ -300,6 +300,100 @@ builder.declare({
   return res.reply(result);
 });
 
+/** Get task group info */
+builder.declare({
+  method: 'get',
+  route: '/task-group/:taskGroupId',
+  name: 'getTaskGroup',
+  scopes: 'queue:list-task-group:<taskGroupId>',
+  stability: APIBuilder.stability.stable,
+  category: 'Tasks',
+  output: 'task-group-response.yml',
+  title: 'Get Task Group',
+  description: [
+    'Get task group information by `taskGroupId`.',
+    '',
+    'This will return meta-information associated with the task group.',
+    'It contains information about task group expiry date or if it is sealed.',
+  ].join('\n'),
+}, async function (req, res) {
+  let taskGroupId = req.params.taskGroupId;
+
+  const taskGroup = await TaskGroup.get(this.db, taskGroupId);
+  if (!taskGroup) {
+    return res.reportError('ResourceNotFound',
+      'No task-group with taskGroupId: `{{taskGroupId}}`', {
+        taskGroupId,
+      });
+  }
+
+  // fetch project ids to construct scopes: `queue:seal-task-group:<taskGroupId>`
+  let projectIds = await taskGroup.getProjectIds(this.db);
+
+  await req.authorize({
+    taskGroupId,
+    projectIds,
```

Comment:
How do the projectIds relate to the scopes here? Doesn't the caller just need "queue:list-task-group:<taskGroupId>" ?

Reply:
Have a look at [req.authorize](https://github.com/taskcluster/taskcluster/blob/2e992676afd1035d044349b7f50926e97675e499/libraries/api/src/middleware/auth.js#L42-L89) - I don't think you need to call `req.authorize` at all, but I am not deeply familiar with this middleware.

Reply:
🤦 thanks pete.. I copy-pasted too much.. this is a different endpoint, it shouldn't be here.
👍